src/renderer/components/inputs/InputContainer.tsx

```diff
@@ -108,12 +108,20 @@ export const InputContainer = memo(
 
         const parentTypeColor = useMemo(() => {
             if (connectedEdge) {
-                const parentNode: Node<NodeData> = getNode(connectedEdge.source)! as Node<NodeData>;
+                const parentNode: Node<NodeData> | undefined = getNode(connectedEdge.source);
                 const parentOutputId = parseSourceHandle(connectedEdge.sourceHandle!).inOutId;
-                const parentType = functionDefinitions
-                    .get(parentNode.data.schemaId)!
-                    .outputDefaults.get(parentOutputId)!;
-                return getTypeAccentColors(parentType, typeDefinitions, isDarkMode)[0];
+                if (parentNode) {
+                    const parentDef = functionDefinitions.get(parentNode.data.schemaId);
+                    if (!parentDef) {
+                        return defaultColor;
+                    }
+                    const parentType = parentDef.outputDefaults.get(parentOutputId);
+                    if (!parentType) {
+                        return defaultColor;
+                    }
+                    return getTypeAccentColors(parentType, typeDefinitions, isDarkMode)[0];
+                }
+                return getTypeAccentColors(NeverType.instance, typeDefinitions, isDarkMode)[0];
```

Comment:
This should have been another `return defaultColor;` but whatever.

Reply:
Ah shit. Didn't notice it because of the formatting

Reply:
I knew I should have left this open for another check but I was confident I had fixed everything... Oh well